deploy.sh

```diff
@@ -69,98 +116,339 @@ if [ $? -eq 0 ]; then
+        downloadNewArchives
+      fi
+      ;;
+
+    -u | --update)
+      # Update option has been selected.
+      if [[ "$2" == "all" ]]; then
+        echo "Selected 'all'. All previous archives will be checked. if inconsistent with remote version, new file will be downloaded."
+        updateAllArchives
+        ((i++));
+        shift
+      else
+        echo "Switching to Default behavior: Only database archives will be updated."
+        updateDataArchives
+        echo "Archives updated"
+      fi
+      ;;
+
+    -b | --build)
+      # This is build option. Used to build webapps for tomcat
+      declare -a app_list=(CuratorTool PathwayExchange RESTfulAPI PathwayBrowser SearchCore DataContent ContentService AnalysisToolsCore AnalysisToolsService AnalysisBin InteractorsCore)
+      if [[ "$2" == "all" ]]; then
+        echo "Selected all: These are all webapps which will be built:"
+        for app_name in "${app_list[@]}"; do
+          echo "${app_name}"
+          export "state_${app_name}=develop"
+        done
+        ((i++));
+        shift
+      elif [[ "$2" == "select" ]]; then
+        echo "Please select which applications you want to build: Press [y/n]"
+        for app_name in "${app_list[@]}"; do
+          echo
+          read -p "${app_name}?`echo $'\n> '`" -n 1 -r
+          if [[ $REPLY =~ ^[Yy]$ ]]; then
+            export "state_${app_name}=develop"
+          else export "state_${app_name}=ready"
+          fi
+        done
+        # Using shift to pop out 'select' argument
+        ((i++));
+        shift
+      else
+        echo "Default behavior: Only essential applications will be built"
+        echo "Essential applications: ReactomeRESTfulAPI.war"
+        export state_RESTfulAPI=develop
+        echo "                        PathwayBrowser.war"
+        export state_PathwayBrowser=develop
+        echo "                        analysis-service.war"
+        export state_AnalysisToolsService=develop
+        echo "                        ContentService.war"
+        export state_ContentService=develop
+        echo "                        content.war"
+        export state_DataContent=develop
+      fi
+      # Tell user whatever is going to happen next
+      sleep 1
+      # At this point we have determined which apps we want to build
+      bash java-application-builder/build_webapps.sh |& tee logs/build_webapps.log
+      ;;
 
-cd ../..
-read -p "Build webapps? Press [y/n]" -n 1 -r
-if [[ $REPLY =~ ^[Yy]$ ]]
-then
-  echo
-  echo "========================================================================="
-  echo "                Building webapps using reactome-app-builder"
-  echo "========================================================================="
-  cd ./java-application-builder
-  bash ./build_webapps.sh |& tee ../logs/build_webapps.log
-  cd ..
-  echo "Reactome-app-builder exits here."
+    -h | --help )
+      # Displaying help
+      echo "$usage"
+      exit 0
+      ;;
+    -run )
```

Comment:
Could this be changed to `-r | --run`? Also needs to be documented in the help text.